### PR TITLE
openapi-v3

### PR DESCRIPTION
**Pull request by david-leifker**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: entity-registry/src/main/java/com/linkedin/metadata/models/registry/MergedEntityRegistry.java [hunk 1]
```diff
@@ -92,6 +92,10 @@ public MergedEntityRegistry apply(EntityRegistry patchEntityRegistry)
       }
     }
 
+    if (!patchEntityRegistry.getAspectSpecs().isEmpty()) {
```
Comment (RyanHolstien) [inline entity-registry/src/main/java/com/linkedin/metadata/models/registry/MergedEntityRegistry.java:95]: This appears to be duplicated from above, if this needs to run after the merge entity specs block then first one should be removed.

File: entity-registry/src/test/java/com/linkedin/metadata/models/OpenApiSpecBuilderTest.java [hunk 2]
```diff
@@ -108,9 +108,9 @@ public void testOpenApiSpecBuilder() throws Exception {
         Path.of(getClass().getResource("/").getPath(), "open-api.yaml"),
         openapiYaml.getBytes(StandardCharsets.UTF_8));
 
-    assertEquals(openAPI.getComponents().getSchemas().size(), 914);
-    assertEquals(openAPI.getComponents().getParameters().size(), 56);
-    assertEquals(openAPI.getPaths().size(), 102);
+    assertTrue(openAPI.getComponents().getSchemas().size() >= 882);
```
Comment (RyanHolstien) [inline entity-registry/src/test/java/com/linkedin/metadata/models/OpenApiSpecBuilderTest.java:111]: This seems like an odd spot to have varying numbers, I think we'd want to know if new schemas are showing up? Maybe it's not important though if this will change often.

Comment (david-leifker) [reply entity-registry/src/test/java/com/linkedin/metadata/models/OpenApiSpecBuilderTest.java:111]: They would change on every data model update including changes to the entity-registry. The numbers are going to increase fairly constantly unless we start deprecating things. A simple non-zero check would however not be as useful since it might actually be broken for large chunks of the data models so this is trying to detect a regression rather then ensuring perfect accuracy.

File: metadata-service/war/src/main/java/com/linkedin/gms/servlet/OpenAPIServletConfig.java [hunk 3]
```diff
@@ -18,7 +18,8 @@
       "io.datahubproject.openapi.v1",
       "io.datahubproject.openapi.v2",
       "io.datahubproject.openapi.v3",
-      "com.linkedin.gms.factory.timeline"
+      "com.linkedin.gms.factory.timeline",
+      "org.springdoc"
```
Comment (kevin1chun) [inline metadata-service/war/src/main/java/com/linkedin/gms/servlet/OpenAPIServletConfig.java:22]: Curious, why was this necessary?

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/config/SpringWebConfig.java [hunk 4]
```diff
@@ -38,24 +44,25 @@ public class SpringWebConfig implements WebMvcConfigurer {
...
-
-  static {
-    NONDEFAULT_OPENAPI_PACKAGES = new HashSet<>();
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(OPERATIONS_PACKAGES);
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(SCHEMA_REGISTRY_PACKAGES);
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(OPENLINEAGE_PACKAGES);
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(V1_PACKAGES);
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(V2_PACKAGES);
-    NONDEFAULT_OPENAPI_PACKAGES.addAll(V3_PACKAGES);
-  }
```
Comment (kevin1chun) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/config/SpringWebConfig.java:43]: the new default openapi page will now have all of the above packages right? AKA we don't want to hide them anymore

File: metadata-service/war/src/main/java/com/linkedin/gms/servlet/OpenAPIServletConfig.java (hunk 3 above)
Comment (david-leifker) [reply metadata-service/war/src/main/java/com/linkedin/gms/servlet/OpenAPIServletConfig.java:22]: The `org.springdoc` package includes the `@Controller` classes that create the swagger UI endpoints. Without Spring able to detect those controllers they never get wired up.

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/config/SpringWebConfig.java (hunk 4 above)
Comment (david-leifker) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/config/SpringWebConfig.java:43]: The difference is there is actually no longer a catch-all group. If new endpoints are added they must either fall into an existing group or a new group can be added. This is more explicit and avoids wiring up endpoints inadvertently with some misplaced annotations.

**Comment by david-leifker** [comment]
The python tests are failing for some reason, will be resolved with https://github.com/datahub-project/datahub/pull/10318

**Review by RyanHolstien — APPROVED** [review]
